src/backend.c

```diff
@@ -440,28 +440,37 @@ static bool opencl_test_instruction (hashcat_ctx_t *hashcat_ctx, cl_context cont
 
 static bool read_kernel_binary (hashcat_ctx_t *hashcat_ctx, const char *kernel_file, size_t *kernel_lengths, char **kernel_sources, const bool force_recompile)
 {
-  FILE *fp = fopen (kernel_file, "rb");
+  HCFILE fp;
+  bool is_gzip = false;
 
-  if (fp != NULL)
+  if (hc_fopen (&fp, kernel_file, "rb", false) != false)
   {
     struct stat st;
 
     if (stat (kernel_file, &st))
     {
-      fclose (fp);
+      hc_fclose (&fp);
 
       return false;
     }
 
+    is_gzip = fp.is_gzip;
+
     #define EXTRASZ 100
 
-    char *buf = (char *) hcmalloc (st.st_size + 1 + EXTRASZ);
+    size_t klen = st.st_size;
+
+    if (is_gzip) klen *= 10; // must be >= of uncompress len
```

Comment:
Please remove for writing files compressed completely. Having kernels in readable format helps debugging a lot, especially on nvidia.

You can also get rid of this:

```
  if (is_gzip) klen *= 10; // must be >= of uncompress len
````

Which is super ugly.

Reply:
I know :) ok